junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DefaultLauncher.java

```diff
@@ -80,9 +80,10 @@ public TestPlan discover(LauncherDiscoveryRequest discoveryRequest) {
 	}
 
 	@Override
-	public void execute(LauncherDiscoveryRequest discoveryRequest) {
+	public void execute(LauncherDiscoveryRequest discoveryRequest, TestExecutionListener... listeners) {
 		Preconditions.notNull(discoveryRequest, "LauncherDiscoveryRequest must not be null");
-		execute(discoverRoot(discoveryRequest, "execution"), discoveryRequest.getConfigurationParameters());
+		Preconditions.notNull(listeners, "TestExecutionListener array must not be null");
```

Comment:
You'll want to use `Preconditions.containsNoNullElements(T[], String)` here as well.